app/assets/javascripts/spree/frontend/affirm_checkout.js

```diff
@@ -0,0 +1,74 @@
+(function(){
+  /* only include this setup once */
+  if (!window.AffirmPaymentMethods) {
+
+    /*****************************************************\
+        set the shared checkout data
+    \*****************************************************/
+    affirm.checkout($("#affirm_checkout_payload").data("affirm"));
+
+
+    /* wait for the DOM to be ready */
+    affirm.ui.ready(function(){
+      $(function() {
+
+        /*****************************************************\
+            setup loading and cancel events for the form
+        \*****************************************************/
+        affirm.checkout.on("cancel", function(){
+          $("#checkout_form_payment input.disabled")
```

Comment:
Sure.. see previous. We still have to come up with something that is testable and extendable. Love to discus options here, this is a part that is lacking in Solidus core right now. No clear way to make this general enough without assuming some conventions that are now in solidus frontend.